includes/Database_Upgrader.php

```diff
@@ -186,6 +187,33 @@ protected function remove_broken_text_styles() {
 		update_option( Stories_Controller::STYLE_PRESETS_OPTION, $updated_style_presets );
 	}
 
+	/**
+	 * Migration for version 2.0.3.
+	 * Color presets: Removes fillColor and textColor and unifies to one color.
+	 *
+	 * @return void
+	 */
+	protected function unify_color_presets() {
```

Comment:
Hmm, the tests are now failing for DB Upgrader since the upgrades always run through all the scripts when running `init()` meaning that the results seem to be combined for all migrations combined and not just for one migration.

Since `textColors` is removed within this PR then the previous tests fail.

@swissspidy Thoughts? Is there a way to define the `'WEBSTORIES_DB_VERSION'` per test?

Reply:
Hmm good question. Let me take a look.

/cc @spacedmonkey

Reply:
Should each test just do `add_option( \Google\Web_Stories\Database_Upgrader::PREVIOUS_OPTION, 'VERSION_BEFORE_THIS_PARTICULAR_MIGRATION' );`?

Reply:
We could filter the value of the option using `pre_option_web_stories_db_version` filter?

Reply:
> edit: although that would run later migrations too...

Yep, unfortunately.

> We could filter the value of the option using `pre_option_web_stories_db_version` filter?

The version would need to change before migration and right after the migration -- to prevent the following migrations running. Filtering with `pre_option_...` would still run the later migrations as well.

For testing purposes, we'd need only one single migration to work.

We could for example:
- Allow filtering the `$routines` (we could then remove all but the relevant routine while testing)
- Add a filter for the version where the version is checked per each routine, e.g. into `run_upgrade_routine` (We could then make the version mismatch for the routines that we're not testing)
- Make `run_upgrade_routine` public and allow running the migrations for a certain version.

Any other ideas? Thoughts?

Reply:
I'd simply use reflection to make the individual routines public during tests, and call that routine directly instead of `init` or `run_upgrade_routine`.

Can copy https://github.com/ampproject/amp-wp/blob/1c5d7b23b018142d58caea37c223facf17953905/tests/php/src/PrivateAccess.php to `tests/phpunit/includes` for that

Reply:
Sounds much better, will check 👍

Reply:
Can you just force the database version to `2.0.2` before the test runs?

Reply:
I would also just reset the database version to not after every test for good messure.

Reply:
> Can you just force the database version to 2.0.2 before the test runs? 

`init` runs all the tests since version X. This means that it would run all the migrations since 2.0.2 if new ones are added, not just this one. We need a way to test only one migration at a time.

This needs to be changed for the previously existing tests, too.

E.g. if for the first test we force the version to be 2.0.0. then 2.0.1 and 2.0.2 would also run -- it's still not reliable.